examples/introductory_examples/example_grids.py

```diff
@@ -0,0 +1,405 @@
+# Generate simulated time data with three sound sources
+pa = ac.demo.create_three_sources_2d(mg, h5savefile='')
+# To prepare for beamforming, we need to calculate the frequency spectra (cross spectral matrix)
+# of the signals. For this we'll use the PowerSpectra object.
+ps = ac.PowerSpectra(source=pa, block_size=128, window='Hanning')
+# Next, we'll create a SteeringVector object.
+# Steering vectors are crucial for beamforming; they handle the time delays between microphones.
+# Keep in mind that the steering vector requires a grid as input, which we'll provide later.
+st = ac.SteeringVector(mics=mg)
+# Finally, well instantiate a BeamformerBase object.
+# This is the foundation for our beamforming algorithms,
+# and it takes the frequency data and steering vectors as input.
+bb = ac.BeamformerBase(freq_data=ps, steer=st)
+
+# %%
+# ================
+# Rectangular Grid
+# ================
+#
+# .. _rectangular_grid_example:
+#
+# The ``RectGrid`` class provides a 2D Cartesian grid for beamforming.
+# It's defined by its boundaries in the x-y plane and a constant z-coordinate.
+
+# %%
+# **Basic RectGrid usage**
+#
+# Create a rectangular grid in the x-y plane at z = -0.2 m
+rg = ac.RectGrid(x_min=-0.2, x_max=0.2, y_min=-0.2, y_max=0.2, z=-0.2, increment=0.01)
+
+# %%
+# **Grid Properties**
+#
+# RectGrid provides several useful properties for accessing grid information:
+
+print(f'Grid size: {rg.size}')  # Total number of grid points
+print(f'Number of x steps: {rg.nxsteps}')  # Grid points along the x axis
+print(f'Number of y steps: {rg.nysteps}')  # Grid points along the y axis
+print(f'Grid shape: {rg.shape}')  # (nxsteps, nysteps)
+print(f'Grid positions:\n{rg.pos[:, :5]}')  # First 5 grid point positions
+
+# %%
+# **Grid Extent**
+#
+# The ``extend`` method returns the grid boundaries in a format suitable for
+# matplotlib's :func:`~matplotlib.pyplot.imshow` function.
+
+print(f'Grid extent: {rg.extend()}')
+
+# %%
+# **Beamforming with Rectangular Grids**
+#
+# Let's demonstrate how grid resolution affects beamforming results by varying the grid increment.
+# Before we do this, we need to assign the grid to the steering vector.
+
+st.grid = rg  # Assign the grid to the steering vector
+
+fig = plt.figure(figsize=(12, 5))
+
+# Calculate and plot results for three different grid resolutions
+for i in range(2):
+    rg.increment = 0.01 * (i + 1)  # Increase grid increment
+
+    # Calculate beamforming output for exactly 8 kHz
+    pm = bb.synthetic(8000, 0)
+    Lm = ac.L_p(pm)
+
+    # Plot the results
+    ax = fig.add_subplot(1, 2, i + 1)
+    ax.imshow(
+        Lm.T,
+        origin='lower',
+        vmin=Lm.max() - 10,
+        extent=rg.extend(),
+    )
+    ax.set_title(f'Beamforming Results\n(grid increment: {rg.increment})')
+    ax.set_xlabel('x / m')
+    ax.set_ylabel('y / m')
+
+plt.tight_layout()
+plt.show()
+
+# %%
+# Note how the results become less precise as the grid increment increases.
```

Comment:
We should mention that the runtime increases as the grid increment decreases, so it is a trade off.